src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -1341,31 +1340,31 @@ public enum Dataset
         /// <summary>
         /// The options for the <see cref="ImageClassificationTransformer"/>.
         /// </summary>
-        internal sealed class Options : TransformInputBase
+        public sealed class Options : TransformInputBase
         {
-            /// <summary>
-            /// Location of the TensorFlow model.
-            /// </summary>
-            [Argument(ArgumentType.Required, HelpText = "TensorFlow model used by the transform. Please see https://www.tensorflow.org/mobile/prepare_models for more details.", SortOrder = 0)]
-            public string ModelLocation;
-
             /// <summary>
             /// The names of the model inputs.
             /// </summary>
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 1)]
+            [Argument(ArgumentType.Multiple , HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 1)]
             public string[] InputColumns;
 
             /// <summary>
             /// The names of the requested model outputs.
             /// </summary>
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 2)]
+            [Argument(ArgumentType.Multiple , HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 2)]
             public string[] OutputColumns;
 
+            /// <summary>
+            /// The names of the model input features.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "features", SortOrder = 1)]
+            public string FeaturesColumnName;
```

Comment:
Why do we have both `InputColumns` and `FeaturesColumnName`? We should only have 1.

Reply:
The same applies for `LabelColumnName`.

And also for `OutputColumns` and `ScoreColumnName`.

---
In reply to: [337694012](https://github.com/dotnet/machinelearning/pull/4362#discussion_r337694012) [](ancestors = 337694012)

Reply:
OutputColumns is an array of ScoreColumn and PredictedLabelColumn. Users can have a choice of setting either one or both of these names. I could make OutputColumns internal if that's cleaner.

---
In reply to: [337694812](https://github.com/dotnet/machinelearning/pull/4362#discussion_r337694812) [](ancestors = 337694812,337694012)

Reply:
Would this API take more than one input at any point, as InputColumn is an array? I could either make InputColumn internal or get rid of one of InputColumns and FeaturesColumnName.

---
In reply to: [337694012](https://github.com/dotnet/machinelearning/pull/4362#discussion_r337694012) [](ancestors = 337694012)

Reply:
Making InputColumns and OutputColumns internal after talking with @eric.erhardt@microsoft.com offline.

---
In reply to: [337712848](https://github.com/dotnet/machinelearning/pull/4362#discussion_r337712848) [](ancestors = 337712848,337694012)